services/ingestion/engine.go

```diff
@@ -140,6 +142,14 @@ func (e *Engine) processEvents(events flow.BlockEvents) error {
 		}
 	}
 
+	// this is an optimization, because if there is a block event in batch, it will internally update latest height
```

Comment:
Good catch :+1: 
I think we can better handle this when we make use of the `grpc.BaseClient` and provide a heartbeat interval of `1`. That way, we'll get back a response for each Cadence block. It may not have EVM events, but at least we can record the Cadence height with more certainty.

Reply:
As an example, see: https://github.com/m-Peter/flow-evm-previewnet/blob/main/main.go#L23-L45
I used `grpc.WithHeartbeatInterval(20)`, but it is safer to have `grpc.WithHeartbeatInterval(1)`.

Reply:
I would say it goes both ways. If we have interval of 1 we have a lot of network noise and updates in the database and I don't think it's worth it. A good balance should be struck. Because restarts are not expected to happen often and when they do it takes really fast to process missed blocks (~100 blocks/s) so it's not worth to have a lot of traffic all the time for sake of very rarely speed up the sync and optimize couple of seconds if you know what I mean.